### PR TITLE
Change address order

### PR DESCRIPTION
**Pull request by whpearson**
Changing the order of address lookup
    
    For this story there is a few aspects. We want to do more accurate uprn
    lookups for eligibility. This requires the following
    
    1) Doing the address lookup before asking for the nhs letter.
    2) No longer doing the postcode to ladcode lookup in browser (instead
       use new functionality in the databse to perform the uprn and postcode
    to tier lookups directly)
    3) Changing all the tests to get the right order.

### CODE REVIEW

File: loadtests/src/test/scala/svp/ShieldedVulnerablePeople.scala [hunk 1]
```diff
@@ -42,6 +42,13 @@ object ShieldedVulnerablePeople {
     .check(status.is(302)))
     .pause(Config.pauseMin, Config.pauseMax)
 
+  private val reqAddressLookupPost = exec(http("POST - Address lookup")
```
Comment (sanisc) [inline loadtests/src/test/scala/svp/ShieldedVulnerablePeople.scala:45]: this just looks like it has moved in the file? I don't see a change in the simulation file to accommodate the changes in the UI

**Review by sanisc — CHANGES_REQUESTED** [review]

File: vulnerable_people_form/form_pages/shared/location_tier.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+from vulnerable_people_form.integrations.location_eligibility import get_uprn_tier, get_postcode_tier, \
+                                                                     is_postcode_in_england
+
+
+def update_is_postcode_in_england(postcode, app):
+    postcode_in_england = is_postcode_in_england(postcode)
+    set_is_postcode_in_england(postcode_in_england)
+
+
+def update_location_tier_for_uprn(uprn, app):
```
Comment (sanisc) [inline vulnerable_people_form/form_pages/shared/location_tier.py:12]: maybe change the naming to update_location_tier_by_uprn

File: vulnerable_people_form/form_pages/shared/location_tier.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    if app.is_tiering_logic_enabled:
+        location_tier = get_uprn_tier(uprn)
+        set_location_tier(location_tier)
+    else:
+        set_location_tier(PostcodeTier.VERY_HIGH_PLUS_SHIELDING.value)
+
+
+def update_location_tier_for_postcode(postcode, app):
```
Comment (sanisc) [inline vulnerable_people_form/form_pages/shared/location_tier.py:20]: same as above

File: vulnerable_people_form/form_pages/address_lookup.py [hunk 4]
```diff
@@ -42,7 +43,7 @@ def get_address_lookup():
         }
         return redirect("/support-address")
 
-    prev_path = append_querystring_params("/nhs-letter" if is_nhs_login_user() else "/date-of-birth")
+    prev_path = append_querystring_params("/postcode-eligibility")
```
Comment (sanisc) [inline vulnerable_people_form/form_pages/address_lookup.py:46]: has this been tested using an nhs login account to make sure the back behaviour works as expected?

Comment (whpearson) [reply vulnerable_people_form/form_pages/address_lookup.py:46]: Ah, no I'll try and get that set up locally.

File: behave/features/10.e2e_journey_tier_3_manual_address.feature [hunk 5]
```diff
@@ -42,26 +56,12 @@ Feature: COVID-19 Shielded vulnerable people service - basic e2e user journey -
         And I submit the form
         Then I am redirected to the "date-of-birth" page
 
-    Scenario: Should be re-directed to address lookup when valid date of birth entered
+    Scenario: Should be re-directed to shopping assistance you when valid date of birth entered
```
Comment (thomasridd) [inline behave/features/10.e2e_journey_tier_3_manual_address.feature:59]: Comment - non-blocking: `you` needs removing

File: vulnerable_people_form/form_pages/shared/routing.py [hunk 6]
```diff
@@ -262,70 +272,77 @@ def dynamic_back_url(default="/"):
...
-        if latest_postcode_tier == PostcodeTier.VERY_HIGH:
+        _raise_returning_user_redirect_error(location_tier_change_status, original_location_tier, latest_location_tier)
+    elif location_tier_change_status == PostcodeTierStatus.DECREASED:
+        if latest_location_tier == PostcodeTier.VERY_HIGH:
             return get_redirect_to_terminal_page()
-        elif not is_tier_very_high_or_above(latest_postcode_tier):
+        elif not is_tier_very_high_or_above(latest_location_tier):
             return redirect("/not-eligible-postcode-returning-user")
-        _raise_returning_user_redirect_error(postcode_tier_change_status, original_postcode_tier, latest_postcode_tier)
+        _raise_returning_user_redirect_error(location_tier_change_status, original_location_tier, latest_location_tier)
```
Comment (thomasridd) [inline vulnerable_people_form/form_pages/shared/routing.py:330]: I'm slightly concerned about all this nested conditional logic at a time when new levels are being introduced but that can easily argue for getting this in asap

**Review by thomasridd — APPROVED** [review]
Code seems good. Tested in practice and is working. My only concerns are over the new tiers being added but it sounds better to get this in ASAP and then attack those when the policy is solid

**Review by sanisc — APPROVED** [review]

**Pull request merged**